src/grelu/data/dataset.py

```diff
@@ -1011,3 +1012,250 @@ def __getitem__(self, idx: int, return_compressed=False) -> Tensor:
 
             # One-hot encode
             return indices_to_one_hot(seq)
+
+
+class SpacingMarginalizeDataset(Dataset):
+    """
+    Dataset class to perform pairwise motif distance analysis. One motif
+    is inserted at a fixed position in shuffled sequences and the second motif is
+    inserted at variable distances from the first.
+
+    Args:
+        seqs: DNA sequences as intervals, strings, integer encoded or one-hot encoded.
+        fixed_pattern: A subsequence to insert in the center of each background sequence.
+        variable_pattern: A subsequence to insert into the background sequences at
```

Comment:
`variable_pattern` is a bit misleading  as it suggested the pattern itself is changing-- maybe `moving_pattern`?

Reply:
good idea

Reply:
done